DistroLauncher/algorithms.h

```diff
@@ -130,3 +130,36 @@ template <typename T, typename... Args> void push_back_many(std::vector<T>& vec,
     vec.reserve(vec.size() + sizeof...(args));
     (vec.push_back(std::forward<Args>(args)), ...);
 }
+
+// Follows the semantics of std::getline(stream&, string&) while implemented for an iterator instead of the stream
+// itself, i.e. reads from [input] iterator until [delim] is found. Although it doesn't go to the string, it's
+// extracted from the underlying stream pointed by the [input] iterator. [str] is also erased before insertion.
+// Iterator is taken by copy, as all algorithms accepting iterators are, but one must remember that the underlying
+// stream linked to the iterator will still have it's internal state changed as the iterator advances, i.e.
+// contents are going to be extracted.
+template <class CharT, class Traits, class Allocator>
+std::istreambuf_iterator<CharT, Traits> getline(std::istreambuf_iterator<CharT, Traits> input,
+                                                std::basic_string<CharT, Traits, Allocator>& str, CharT delim = '\n')
+{
+    str.erase();
+    // missing std::copy_until - a copy_if that stops when the predicate evaluates to true :(
```

Comment:
This would be the way if you want to be an algorithm fanboy 😆.
```c++
const auto end = std::find(input, istreambuf_iterator{}, delim);
std::copy(input, end, std::ostream_iterator(str));
return end;
```

Reply:
`iostream`s and their iterators are supposed to be single pass. So that would require remembering the stream position at start and `seek` back to it prior to calling std::copy (raw calls to the `istream` or the `istreambuf`), otherwise `input` and `end` might both end up in the end of the stream if `delim` is not found or `input` becomes one past `end`, if `delim` is found. Interleaving `tellg` and `seek` calls would fix the issue, but then we read the file twice 😞 : (unless compiler does some optimization to inline both algorithms and interleave them, which I don't doubt that may happen in optimized builds, but not in debug and it's hard to rely on that fact).

_"No raw loops"_ not always work.